sw/host/ot_certs/src/x509/tests.rs

```diff
@@ -0,0 +1,571 @@
+        compare_bigint,
+    );
+    compare_opt("DICE index", &None, rand, &dice_tcb.index, compare_bigint);
+    assert_eq!(
+        dice_tcb.index, None,
+        "DICE index should not contain a value but is {:?}",
+        dice_tcb.index
+    );
+    assert_eq!(
+        template.certificate.flags, dice_tcb.flags,
+        "DICE flags mismatch"
+    );
+    assert_eq!(
+        dice_tcb.vendor_info, None,
+        "DICE TCB vendor information should not contain a value but is {:?}",
+        dice_tcb.index
+    );
+    assert_eq!(
+        dice_tcb.tcb_type, None,
+        "DICE TCB type information should not contain a value but is {:?}",
+        dice_tcb.tcb_type
+    );
+    // Public key.
+    let pub_key = x509
+        .public_key()
+        .expect("the X509 does not have a valid public key!");
+    let ec_pub_key = EcKey::<Public>::try_from(pub_key).expect("the X509 is not an EC public key!");
+    let mut ctx = BigNumContext::new().unwrap();
+    let mut x = BigNum::new().unwrap();
+    let mut y = BigNum::new().unwrap();
+    ec_pub_key
+        .public_key()
+        .affine_coordinates(ec_pub_key.group(), &mut x, &mut y, &mut ctx)
+        .unwrap();
+    // Convert x and y to the DER representation, potentially adding some padding if necessary.
+    let SubjectPublicKeyInfo::EcPublicKey(EcPublicKeyInfo { public_key, .. }) =
+        &template.certificate.subject_public_key_info;
+    compare_integer(
+        "public key EC x",
+        &public_key.x,
+        rand,
+        &x.to_asn1_integer().unwrap(),
+    );
+    compare_integer(
+        "public key EC y",
+        &public_key.y,
+        rand,
+        &y.to_asn1_integer().unwrap(),
+    );
+}
+
+fn compare_names(
+    field: &str,
+    name: &HashMap<AttributeType, Value<String>>,
+    rand: &RandData,
+    x509_name: &X509NameRef,
+) {
+    println!("compare {:?} with {:?}", name, x509_name);
+    assert_eq!(
+        name.len(),
+        x509_name.entries().count(),
+        "{} field does not have the right number of entries",
+        field
+    );
+    for (attr, val) in name {
+        let nid_str = attr.nid().long_name().unwrap_or("????");
+
+        let x509_name = x509_name.to_owned().expect("cannot copy x509 name");
+        let mut entries = x509_name.entries_by_nid(attr.nid());
+        let entry = entries.next().unwrap_or_else(|| {
+            panic!(
+                "{} field has no {} entry in the X509 but got none",
+                field, nid_str
+            )
+        });
+        assert_eq!(
+            entries.next().is_none(),
+            true,
+            "{} field has more than one {} entry in the X509",
+            field,
+            nid_str
+        );
+        compare_asn1_string(
+            &format!("{} field entry {}", field, nid_str),
+            val,
+            rand,
+            entry.data(),
+        );
+    }
+}
+
+fn filter<T: std::fmt::Debug>(rand: &HashMap<String, T>, expect: &Value<T>) -> Option<String> {
```

Comment:
The name feels non-descriptive for what it's doing.

Reply:
I removed it, it was only used for debug.